pkg/httpserver/server.go

```diff
@@ -115,16 +145,54 @@ func (s *Server) Stop(ctx context.Context) error {
 }
 
 type healthCheckResp struct {
-	Status      string    `json:"status"`
-	CurrentTime time.Time `json:"currentTime"`
+	MQStatus    string    `json:"mqStatus,omitempty"`
+	VCTStatus   string    `json:"vctStatus,omitempty"`
+	DBStatus    string    `json:"dbStatus,omitempty"`
+	CurrentTime time.Time `json:"currentTime,omitempty"`
+	Version     string    `json:"version,omitempty"`
 }
 
-func healthCheckHandler(rw http.ResponseWriter, r *http.Request) {
-	rw.WriteHeader(http.StatusOK)
+func (s *Server) healthCheckHandler(rw http.ResponseWriter, r *http.Request) { //nolint:gocyclo,cyclop
+	mqStatus := ""
+	vctStatus := ""
+	dbStatus := ""
+
+	if s.pubSub != nil {
+		mqStatus = success
+
+		if err := s.pubSub.IsConnected(); err != nil {
+			mqStatus = err.Error()
+		}
+	}
+
+	if s.vct != nil {
+		vctStatus = success
+
+		if err := s.vct.HealthCheck(); err != nil {
+			vctStatus = err.Error()
+		}
+	}
+
+	if s.db != nil {
+		dbStatus = success
+
+		if err := s.db.Ping(); err != nil {
+			dbStatus = err.Error()
+		}
+	}
+
+	if mqStatus != success || vctStatus != success || dbStatus != success {
+		rw.WriteHeader(http.StatusInternalServerError)
```

Comment:
Maybe 503 - Service Unavailable would be more appropriate?

Reply:
ok

Reply:
done